src/framework/MOM_restart.F90

```diff
@@ -493,17 +514,36 @@ subroutine register_restart_field_3d(f_ptr, name, mandatory, CS, longname, units
   character(len=*), optional, intent(in) :: hor_grid  !< variable horizontal staggering, 'h' if absent
   character(len=*), optional, intent(in) :: z_grid    !< variable vertical staggering, 'L' if absent
   character(len=*), optional, intent(in) :: t_grid    !< time description: s, p, or 1, 's' if absent
+  type(axis_info),  dimension(:), &
+                    optional, intent(in) :: extra_axes !< dimensions other than space-time
 
   type(vardesc) :: vd
+  character(len=32), dimension(:), allocatable :: dim_names
+  integer :: n, n_extradims
+
+  if (present(extra_axes)) then
```

Comment:
Add a comment describing the padding, or referencing the previous comment (see above)